lib/shopify_app/scripttags_manager.rb

```diff
@@ -57,5 +60,11 @@ def scripttag_exists?(src)
     def current_scripttags
       @current_scripttags ||= ShopifyAPI::ScriptTag.all.index_by(&:src)
     end
+
+    def build_src(src)
+      src.respond_to?(:call) ? src.call(shop_domain) : src
+    rescue => e
+      raise CreationFailed, "cannot create script src: #{e.message}"
```

Comment:
If we do it right then figuring out what happened shouldn't be a problem. All that's missing is the original backtrace, which can be solved via:

```ruby
raise CreationFailed, "cannot create script src: #{e.message}", e.backtrace
```

Providing the caller with consistent exception(s) has merit.

I made the change you've requested here, but this brings me back to an original exception related concern I had:

> under what circumstances [does this get called](https://github.com/Shopify/shopify_app/blob/076a2966823cfa2d74179cda675ff942de707c7a/lib/shopify_app/scripttags_manager.rb#L49)? If the API cannot create a resource an exception is raised.